client.go

```diff
@@ -27,13 +29,10 @@ func (b *Backend) clientMongo(ctx context.Context, s logical.Storage) (*mongodba
 
 	pluginEnv, err := b.system.PluginEnv(ctx)
 	if err != nil {
-		return nil, err
-	}
-	if pluginEnv != nil {
-		client.UserAgent = useragent.PluginString(pluginEnv, "mongodbatlas-secrets")
-	} else {
-		client.UserAgent = useragent.String()
+		b.Logger().Warn("failed to read plugin environment, user-agent will not be set",
```

Comment:
I _think_ we can log a warning and carry on if we can't get the plugin environment information. It only contains version information used in the user-agent string. I could also see the argument for returning the error. Let me know your thoughts!

Reply:
I'm ok with that, though I'm neither an HTTP nor a plugin expert.  Alternatively, I can't see any code paths where this would fail but other RPCs would succeed (again, not an authority here, please don't assume this is correct without verifying.)  In which case, what does it buy us to ignore this error?

Reply:
+1 to returning the error instead of logging a warning. I agree that this would indicate an RPC error that should not be ignored.

Reply:
Done in https://github.com/hashicorp/vault-plugin-secrets-mongodbatlas/pull/29/commits/4aa2cc2b0e9b365e769713341ddd7d1b784413a6.